unstructured_ingest/v2/processes/connectors/zendesk/client.py

```diff
@@ -0,0 +1,369 @@
+import base64
+from dataclasses import dataclass
+from typing import Dict, List
+
+import httpx
+import requests
+
+from unstructured_ingest.v2.errors import ProviderError, UserError
+from unstructured_ingest.v2.logger import logger
+
+
+@dataclass
+class Comment:
+    id: str
+    author_id: str
+    body: str
+    parent_ticket_id: str
+    metadata: dict
+
+
+@dataclass
+class ZendeskTicket:
+    id: str
+    subject: str
+    description: str
+    generated_ts: int
+    metadata: dict
+
+    def __lt__(self, other):
+        return self.id < other.id
+
+
+@dataclass
+class ZendeskArticle:
+    id: str
+    author_id: str
+    title: str
+    content: str
+
+    def __lt__(self, other):
+        return self.id < other.id
+
+
+class ZendeskClient:
+
+    def __init__(self, token: str, subdomain: str, email: str):
+        # should be okay to be blocking.
+        url_to_check = f"https://{subdomain}.zendesk.com/api/v2/groups.json"
+        auth = f"{email}/token", token
+        try:
+            response = requests.get(url_to_check, auth=auth)
+
+            http_code = response.status_code
+
+            if 400 <= http_code < 500:
+                message = (
+                    f"Failed to connect to {url_to_check} using zendesk response"
+                    f"status code {http_code}"
+                )
+                return UserError(message)
+
+            if http_code >= 500:
+                return ProviderError(f"Failed to connect to {url_to_check} using zendesk response")
+
+            if http_code != 200:
```

Comment:
Let the request library raise the error if it's one we didn't explicitly catch by checking the status codes. 
```suggestion
            response.raise_for_status()
```